utils/fetchAssets.ts

```diff
@@ -1,18 +1,52 @@
+import LSP3ProfileMetadata from '@erc725/erc725.js/schemas/LSP3ProfileMetadata.json'
+
 import { AssetRepository } from '@/repositories/asset'
 
+import type { ERC725JSONSchema } from '@erc725/erc725.js'
+
 export const fetchAssets = async (profileAddress: Address) => {
   const { profile } = useProfile(profileAddress)
+  const profileRepo = useRepo(ProfileRepository)
   const assetRepo = useRepo(AssetRepository)
-  const { receivedAssetAddresses, issuedAssetAddresses } = profile.value || {}
+  const { getInstance } = useErc725()
+  const erc725 = getInstance(
+    profileAddress,
+    LSP3ProfileMetadata as ERC725JSONSchema[]
+  )
   const { isLoadingAssets } = storeToRefs(useAppStore())
+  let receivedAssets, issuedAssets
 
+  // we split LSP5 and LSP12 into separate catch blocks so we can handle them individually and even if
+  // one fails, the other is still handled.
   try {
     isLoadingAssets.value = true
 
-    receivedAssetAddresses &&
-      (await assetRepo.loadAssets(receivedAssetAddresses, profileAddress))
-    issuedAssetAddresses &&
-      (await assetRepo.loadAssets(issuedAssetAddresses, profileAddress))
+    try {
+      receivedAssets = (await erc725.fetchData('LSP5ReceivedAssets[]'))?.value
+      assertArray(receivedAssets)
+      assertAddresses(receivedAssets)
+
+      receivedAssets &&
+        (await assetRepo.loadAssets(receivedAssets, profileAddress))
+    } catch (error) {
+      throw error
```

Comment:
The separation makes sense but throwing an error would still not do the next one? Seems problematic

Reply:
Right, I'll fix this. I mainly focused on separation between profile and assets, not between issued/received assets. But that will also be useful.